### PR TITLE
Problem: recent fixes in dependencies are not used

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- cosmos-sdk -> v0.46.4
- ethermint -> main
- ibc-go -> v5.0.1
- add dragonberry ics20 replacement

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,24 +5,24 @@ go 1.18
 require (
 	cosmossdk.io/math v1.0.0-beta.3
 	github.com/armon/go-metrics v0.4.1
-	github.com/cosmos/cosmos-sdk v0.46.2
-	github.com/cosmos/ibc-go/v5 v5.0.0
+	github.com/cosmos/cosmos-sdk v0.46.4
+	github.com/cosmos/ibc-go/v5 v5.0.1
 	github.com/ethereum/go-ethereum v1.10.19
-	github.com/evmos/ethermint v0.6.1-0.20221003153722-491c3da7ebd7
+	github.com/evmos/ethermint v0.6.1-0.20221101220534-a8ea4eceb6d9
```
Comment (yihuang) [inline go.mod:11]: https://github.com/evmos/ethermint/compare/491c3da7ebd7...a8ea4eceb6d9

Comment (yihuang) [reply go.mod:11]: it contains a breaking change, it add support to multisig of ethsecp256k1 keys, maybe ok to smuggle into testnet?

Comment (yihuang) [reply go.mod:11]: switched to a fork, which reverted the ethsecp256k1 multisig support.

File: go.mod [hunk 2]
```diff
@@ -189,8 +189,11 @@ require (
 )
 
 replace (
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.4-0.20221028111015-4eed46eba492
+	// ics23 patch for dragonberry
+	github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.4
 	github.com/ethereum/go-ethereum => github.com/ethereum/go-ethereum v1.10.19
+	github.com/evmos/ethermint => github.com/crypto-org-chain/ethermint v0.20.0-cronos
```
Comment (yihuang) [inline go.mod:196]: https://github.com/crypto-org-chain/ethermint/compare/491c3da7ebd7...v0.20.0-cronos

**Review by thomas-nguy — APPROVED** [review]

**Comment by thomas-nguy** [comment]
quick glance, changes looks non breaking however better to test before releasing

**Review by thomas-nguy — CHANGES_REQUESTED** [review]
https://github.com/evmos/ethermint/pull/1388

actually does this change the gas consumed for failed txs?

**Comment by yihuang** [comment]
> [evmos/ethermint#1388](https://github.com/evmos/ethermint/pull/1388)
> 
> actually does this change the gas consumed for failed txs?

does it? it seems only reorder some codes to make it efficient, don't change behavior?

**Review by thomas-nguy — APPROVED** [review]
Ok this reordering looks okay because the two other handler put at the top seems stateless.

I was thinking that it could affect the gas_used somehow if the order changes

**Comment by yihuang** [comment]
> Ok this reordering looks okay because the two other handler put at the top seems stateless.
> 
> I was thinking that it could affect the gas_used somehow if the order changes

I see, you mean if the tx fail in ante handler, the gas used will be calculated by the sdk operations?

**Comment by thomas-nguy** [comment]
also the sdk.error could also changes depending of the order?

**Comment by yihuang** [comment]
> also the sdk.error returned could also changes depending of the order?

error message is not normally part of the consensus, unless like in the old case, the ibc module record the error message into consensus state, it shouldn't happen now though.

**Review by tomtau — APPROVED** [review]

**Pull request merged**